cmd/localstack/xraydaemon.go

```diff
@@ -235,7 +240,7 @@ func (d *Daemon) poll() {
 			}
 			continue
 		}
-		if fallbackPointerUsed {
+		if fallbackPointerUsed && telemetry.T != nil {
 			log.Warn("Segment dropped. Consider increasing memory limit")
```

Comment:
nit: it would be nice to log dropping segments even if telemetry is disabled

Reply:
also the continue would've been skipped which might have been an issue as well